### PR TITLE
Replace start_dependencies with docker compose --wait

### PR DESCRIPTION
**Pull request by lieut-data**
#### Summary

I'm on a mission to improve our developer tooling. I just merged https://github.com/mattermost/mattermost/pull/35975 to remove the need for a Darwin/arm64 override and reduce the number of Docker Compose files. I'm aiming towards a future where all my worktrees can operate standalone and `docker compose` works out of the box. (There's still a lot of docker compose files in play: I'm explicitly /not/ fixing that with this PR.)

This incremental change eliminates the custom `mattermost-wait-for-dep` container and the Go-based docker-compose generator in favour of `docker compose --wait` and profiles.

As a developer, nothing should change: we still honour the `ENABLED_DOCKER_SERVICES`, but now just translate that into `--profile <service>` parameters automatically. The healthchecks are now inlined as well, most of which are not "full healthchecks" in favour of just testing that the port opens. Startup looks roughly the same as before:

CleanShot 2026-04-08 at 11 26 49

#### Ticket Link
N/A

#### Release Note
```release-note
NONE
```

## Change Impact: 🟡 Medium

**Regression Risk:** Replaces the custom dependency bootstrap with docker compose --wait and profiles, changing CI and developer-local startup/ordering and inlining many healthchecks to simpler port probes; risk is concentrated in environment/boot/readiness detection (including edge cases like the scratch-based otel-collector) but application core/auth/persistence logic is untouched. Regression likelihood is moderate—failures would mainly affect CI/local startup and test harness behavior rather than runtime business functionality.

**QA Recommendation:** Moderate manual validation: run CI and local startup on representative platforms (Linux and macOS Intel/ARM) with several ENABLED_DOCKER_SERVICES combinations, verify services become healthy within the 300s timeout, and execute relevant server/mmctl E2E or integration flows to confirm test and development workflows behave equivalently.

*Generated by CodeRabbitAI*

### CODE REVIEW

File: server/docker-compose.makefile.yml
Comment (lieut-data) [inline server/docker-compose.makefile.yml:1]: I'm hoping to converge to a vanilla `docker-compose.yml` eventually.

File: server/docker-compose.yaml [hunk 1]
```diff
@@ -126,7 +109,24 @@ services:
...
+      elasticsearch:
+        condition: service_healthy
+      opensearch:
+        condition: service_healthy
+      prometheus:
+        condition: service_healthy
+      grafana:
+        condition: service_healthy
+      loki:
+        condition: service_healthy
```
[the remark this answers is not in the document]
Comment (lieut-data) [reply server/docker-compose.yaml:129]: The old `start_dependencies` service in `docker-compose.yaml` didn't include redis either — its `depends_on` list was: postgres, minio, inbucket, openldap, elasticsearch, opensearch, prometheus, grafana, loki, otel-collector. The new `depends_on` on `leader` faithfully mirrors that. Adding redis would be a scope expansion beyond this PR.

**Comment by lieut-data** [comment]
Responding to the remaining CodeRabbit findings:

**otel-collector healthcheck:** The image (`otel/opentelemetry-collector-contrib:0.145.0`) is scratch-based — no shell, no `nc`, `wget`, `curl`, or `bash`. There's no binary available to probe the health endpoint from inside the container. `--wait` treats services without healthchecks as ready once started, which matches the old `wait-for-dep` behavior (it probed from its own container, not from within each service).

**local-test-env.sh syntax errors:** Acknowledged — the missing backslashes and outdated build image are real but pre-existing. Out of scope for this PR.
